apps/condo/domains/miniapp/access/B2BAppRole.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Generated by `createschema miniapp.B2BAppRole 'app:Relationship:B2BApp:CASCADE; role:Relationship:OrganizationEmployeeRole:CASCADE; permissions:Json;'`
+ */
+
+const get = require('lodash/get')
+
+const { throwAuthenticationError } = require('@open-condo/keystone/apolloErrorFormatter')
+const { find, getById } = require('@open-condo/keystone/schema')
+
+const { queryOrganizationEmployeeFor, checkOrganizationPermission } = require('@condo/domains/organization/utils/accessSchema')
+const { SERVICE, STAFF } = require('@condo/domains/user/constants/common')
+
+/**
+ * B2B App permissions may only be read:
+ * 1. Admin / support
+ * 2. Integration service account
+ * 3. Organization employees
+ * 3.a If employee has 'canManageRoles' in role = return all roles from this organization
+ * 3.b Otherwise = return only roles to user employees
+ */
+async function canReadB2BAppRoles ({ authentication: { item: user } }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+
+    if (user.isAdmin || user.isSupport) return {}
+
+    if (user.type === SERVICE) {
+        return { app: { accessRights_some: { user: { id: user.id }, deletedAt: null } } }
+    }
+
+    if (user.type === STAFF) {
+        const employeeCondition = queryOrganizationEmployeeFor(user.id).employees_some
+        const userEmployees = await find('OrganizationEmployee', employeeCondition)
+        const userEmployeesIds = userEmployees.map(employee => employee.id)
+        const userRoleIds = userEmployees.map(employee => employee.role)
+
+        return {
+            role: {
+                OR: [
+                    { id_in: userRoleIds },
+                    {
+                        organization: {
+                            employees_some: { id_in: userEmployeesIds, role: { canManageRoles: true } },
```

Comment:
`deletedAt: null`?

Reply:
It's not necessary. Look what happened few lines above:
1. We've git our classic employee filter (blocked, rejected, deleted and so on)
2. Then we already got all non-deleted user employees.
3. For `non-manager` we are returning only roles for this `employees` fetched above.
4. For `manager` we return all roles in organizations in which he's manager